test/t/unit/test_unit_compgen_filedir.py

```diff
@@ -259,3 +259,16 @@ def test_28_dot_4(self, bash, functions, funcname):
         """Include . when the completion is attempted for '..[TAB]'"""
         completion = assert_complete(bash, r"%s _filedir/.." % funcname)
         assert completion == "/"
+
+    @pytest.mark.parametrize("funcname", "f f2".split())
+    def test_29_dotdot(self, bash, functions, funcname):
+        """Complete files starting with .."""
```

Comment:
I think the explanation of the test case in the commit message is better to be put inside the docstring (separated by an empty line).